tests/tests_geomstats/test_spd_matrices.py

```diff
@@ -19,559 +21,548 @@
+        result = metric.exp_domain(
+            gs.array(tangent_vec), gs.array(base_point), expected
         )
-        metric = self.metric_affine
-        geodesic = metric.geodesic(
-            initial_point=initial_point, initial_tangent_vec=initial_tangent_vec
+        self.assertAllClose(result, gs.array(expected))
+
+    def test_log(self, n, power_euclidean, point, base_point, expected):
+        metric = SPDMetricEuclidean(n)
+        result = metric.log(gs.array(point), gs.array(base_point))
+        self.assertAllClose(result, gs.array(expected))
+
+    def test_log_exp_composition(self, n, power_euclidean, point, base_point):
+        metric = SPDMetricEuclidean(n, power_euclidean)
+        log = metric.log(gs.array(point), base_point=gs.array(base_point))
+        result = metric.exp(tangent_vec=log, base_point=gs.array(base_point))
+        self.assertAllClose(result, point)
+
+
+class TestSPDMetricLogEuclidean(geomstats.tests.TestCase, metaclass=Parametrizer):
+    class TestDataSPDMetricLogEuclidean(TestData):
+        def inner_product_data(self):
+            smoke_data = [
+                dict(
+                    n=3,
+                    tangent_vec_a=[[1.0, 1.0, 3.0], [1.0, 1.0, 3.0], [3.0, 3.0, 4.0]],
+                    tangent_vec_b=[[1.0, 1.0, 3.0], [1.0, 1.0, 3.0], [3.0, 3.0, 4.0]],
+                    base_point=[[1.0, 0.0, 0.0], [0.0, 1.0, 0.0], [0.0, 0.0, 4.0]],
+                    expected=5.0 + (4.0 * ((2 * LN_2) ** 2)),
+                )
+            ]
+            return self.generate_tests(smoke_data)
+
+        def exp_data(self):
+            smoke_data = [
+                dict(
+                    n=2,
+                    tangent_vec=[[2.0, 0.0], [0.0, 2.0]],
+                    base_point=[[1.0, 0.0], [0.0, 1.0]],
+                    expected=[[EXP_2, 0.0], [0.0, EXP_2]],
+                )
+            ]
+            return self.generate_tests(smoke_data)
+
+        def log_data(self):
+            smoke_data = [
+                dict(
+                    n=2,
+                    point=[[2.0, 0.0], [0.0, 2.0]],
+                    base_point=[[1.0, 0.0], [0.0, 1.0]],
+                    expected=[[LN_2, 0.0], [0.0, LN_2]],
+                )
+            ]
+            return self.generate_tests(smoke_data)
+
+        def log_exp_composition_data(self):
+            return self.log_exp_composition(SPDMatrices)
+
+    testing_data = TestDataSPDMetricLogEuclidean()
+
+    def test_inner_product(self, n, tangent_vec_a, tangent_vec_b, base_point, expected):
+        metric = SPDMetricLogEuclidean(n)
+        result = metric.inner_product(
+            gs.array(tangent_vec_a), gs.array(tangent_vec_b), gs.array(base_point)
         )
-
-        n_points = 10
-        t = gs.linspace(start=0.0, stop=1.0, num=n_points)
-        points = geodesic(t)
-        result = self.space.belongs(points)
-        self.assertTrue(gs.all(result))
-
-    def test_squared_dist_is_symmetric(self):
```

Comment:
Could we add `test_squared_dist_is_symmetric` within the randomized tests? I'm afraid we will forget about these tests if not.